raft.go

```diff
@@ -1735,11 +1747,12 @@ type voteResult struct {
 // ourself. This has the side affecting of incrementing the current term. The
 // response channel returned is used to wait for all the responses (including a
 // vote for ourself). This must only be called from the main thread.
-func (r *Raft) electSelf() <-chan *voteResult {
+func (r *Raft) electSelf(preVote bool) <-chan *voteResult {
 	// Create a response channel
 	respCh := make(chan *voteResult, len(r.configurations.latest.Servers))
 
 	// Increment the term
+	// PreVote campaigns are sent on the same term.
```

Comment:
I'm confused, the comment points out that `PreVote` shouldn't increment the term (true) but the code still does increment the term even if `preVote == true`.

Am I reading this wrong? If we are still incrementing the term then I don't think this is a working implementation of `PreVote` because this candidate will still increment it's term as long as it's disconnected and then will still disrupt the cluster with a newer term when it is reconnected right?

Or did I missunderstand?